src/trng.jl

```diff
@@ -0,0 +1,27 @@
+import Random
+using Random: AbstractRNG, MersenneTwister
+
+struct ThreadedRNG{R<:AbstractRNG} <: AbstractRNG
+    rngs::Vector{R}
+    function ThreadedRNG(rngs)
+        len = length(rngs)
+        nt = Threads.nthreads()
+        if len >= nt
+            msg = """length(rngs) >= Threads.nthreads() must hold. Got:
+            length(rngs) = $len
+            Threads.nthreads() = $nt
+            """
+        end
+        new(rngs)
+    end
+end
+
+const GLOBAL_TRNG = ThreadedRNG(map(MersenneTwister, 1:Threads.nthreads()))
+
+for f in [:rand, :randn, :rand!, :randn!]
```

Comment:
Can we have some tests for these please?

Reply:
Ah dang this was an experiment and only included by accident in this PR. It was really late yesterday. I will remove it. By the way I was not satisfied with the experiment, the code gets ugly. You need to overload a lot of signatures, otherwise `rand` and friends will produce disambiguities. See also [here](https://discourse.julialang.org/t/counting-number-of-times-a-rng-is-called/24903/17?u=jw3126)